### PR TITLE
Add app and deployment-group module

### PR DESCRIPTION
**Pull request by MattiasGees**

### CODE REVIEW

**Comment by ringods** [comment]
Why 2 TF modules? If the `aws_codedeploy_app` and `aws_codedeploy_deployment_group` usually go together, why not abstract them in a single module?

**Review by iuriaranda — APPROVED** [review]

**Comment by MattiasGees** [comment]
@ringods what if you have staging and production? 
You only need one codedeploy app for those, but two codedeploy deployment groups.

**Comment by ringods** [comment]
Are the 2 deployment groups then linked to the same app?
(Not up to speed on CodeDeploy, so just asking.. 😃)

File: README.md [hunk 1]
```diff
@@ -1,2 +1,46 @@
...
+
+## app
+Create a codedeploy app
+
+### Available variables
+ * [`name`]: String(required): Name of your codedeploy application
+ * [`project`]: String(required): The current project
+
+### Output
+* [`app_name`]: Sreing: Name of the codedeploy app
```
Comment (SamClinckspoor) [inline README.md]: typo `String`

File: README.md [hunk 2]
```diff
@@ -1,2 +1,46 @@
...
+
+### Output
+/
+
+### Example
+```
+  module "deployment_group" {
+    source             = "github.com/skyscrapers/terraform-codedeploy//deployment-group"
+    environment        = "production"
+    app_name           = "test-app"
```
Comment (SamClinckspoor) [inline README.md]: you could use the output of `module.codedeploy.name` if you want

File: README.md [hunk 3]
```diff
@@ -1,2 +1,46 @@
...
+  }
+```
+
+## deployment_group
+Create an deployment group for a codedeploy app
+
+### Available variables
+ * [`environment`]: String(required): Environment where your codedeploy deployment group is used for
+ * [`app_name`]: String(required): Name of the coddeploy app
+ * [`service_role_arn`]: String(required): IAM role that is used by the deployment group
```
Comment (SamClinckspoor) [inline README.md]: maybe reference the iam module here

**Comment by MattiasGees** [comment]
yes so you have for example a codedeploy application called `java-app`. To that codedeploy you add the deployment groups `java-app-staging` and `java-app-production`. This way you can use codedeploy to easily deploy the same source to all environments.

**Review by SamClinckspoor — APPROVED** [review]

**Pull request merged**